src/UraniumUI.Material/Controls/CheckBox.Windows.cs

```diff
@@ -4,26 +4,20 @@
 namespace UraniumUI.Material.Controls;
 public partial class CheckBox
 {
-    protected override void OnHandlerChanged()
+    protected override void OnHandlerChanging(HandlerChangingEventArgs args)
     {
-        base.OnHandlerChanged();
-        if (Handler != null)
+        base.OnHandlerChanging(args);
+        if (args.OldHandler != null && args.OldHandler.PlatformView is Panel oldContentPanel)
```

Comment:
```suggestion
        if (args.OldHandler?.PlatformView is Panel oldContentPanel)
```

same in the other location